src/Utilities/model.jl

```diff
@@ -1077,130 +996,37 @@ macro model(
             upper_bound::Vector{T}
-            var_to_name::Dict{$VI,String}
+            constraints::C
+            var_to_name::Dict{MOI.VariableIndex,String}
             # If `nothing`, the dictionary hasn't been constructed yet.
-            name_to_var::Union{Dict{String,$VI},Nothing}
-            con_to_name::Dict{$CI,String}
-            name_to_con::Union{Dict{String,$CI},Nothing}
+            name_to_var::Union{Dict{String,MOI.VariableIndex},Nothing}
+            con_to_name::Dict{MOI.ConstraintIndex,String}
+            name_to_con::Union{Dict{String,MOI.ConstraintIndex},Nothing}
             # A useful dictionary for extensions to store things. These are
             # _not_ copied between models!
             ext::Dict{Symbol,Any}
-        end
-    end
-    for f in funs
-        cname = f.cname
-        field = _field(f)
-        push!(modeldef.args[2].args[3].args, :($field::$cname{T,$(_getC(f))}))
-    end
-
-    code = quote
-        function $MOIU.broadcastcall(f::F, model::$esc_model_name) where {F<:Function}
-            return $(Expr(:block, _broadcastfield.(Ref(:(broadcastcall)), funs)...))
-        end
-        function $MOIU.broadcastvcat(f::F, model::$esc_model_name) where {F<:Function}
-            return vcat($(_broadcastfield.(Ref(:(broadcastvcat)), funs)...))
-        end
-        function $MOIU.mapreduce_constraints(f::Function, op::Function, model::$esc_model_name, cur)
-            return $(Expr(:block, _mapreduce_field.(funs)...))
-        end
-    end
-    for (cname, sets) in ((scname, scalar_sets), (vcname, vector_sets))
-        code = quote
-            $code
-            function $MOIU.broadcastcall(f::F, model::$cname) where {F<:Function}
-                return $(Expr(:block, _callfield.(:f, sets)...))
-            end
-            function $MOIU.broadcastvcat(f::F, model::$cname) where {F<:Function}
-                return vcat($(_callfield.(:f, sets)...))
-            end
-            function $MOIU.mapreduce_constraints(f::Function, op::Function, model::$cname, cur)
-                return $(Expr(:block, _mapreduce_constraints.(sets)...))
-            end
-        end
-    end
-
-    for (c, sets) in ((scname, scalar_sets), (vcname, vector_sets))
-        for s in sets
-            set = _set(s)
-            field = _field(s)
-            code = quote
-                $code
-                function $MOIU.constraints(
-                    model::$c,
-                    ::Type{<:$set},
-                ) where {F}
-                    return model.$field
-                end
-            end
-        end
-    end
-
-    for f in funs
-        fun = _fun(f)
-        field = _field(f)
-        code = quote
-            $code
-            function $MOIU.constraints(
-                model::$esc_model_name,
-                ::Type{<:$fun},
-                ::Type{S}
-            ) where S
-                return $MOIU.constraints(model.$field, S)
+            function $name{T,C}() where {T,C}
```

Comment:
So maybe something like:
```julia
Base.show(io::IO, model::$(name){T,C}) where {T,C} = print(io, "$(name){T}")
```

Reply:
What about this show? Do we implement or not?

Reply:
Not. I meant it to be the `Type{}` one.